github/Auth.py

```diff
@@ -190,15 +190,17 @@ def get_installation_auth(
         installation_id: int,
         token_permissions: Optional[Dict[str, str]] = None,
         requester: Optional[Requester] = None,
+        token_repositories: Optional[List[Union[str, int]]] = None,
```

Comment:
Though this method is public, the `requester` argument is internal in a sense that there should be no user code calling this method with `requester` given. Hence, it is safe (and preferable) to add `token_repositories` before `requester`:
```suggestion
        token_permissions: Optional[Dict[str, str]] = None,
        token_repositories: Optional[List[Union[str, int]]] = None,
        *,
        requester: Optional[Requester] = None,
```

Adding the `*` makes the `requester` argument mandatory named, which avoids breaking user code in the future when adding an argument before it.

This requires code that provides `requester` to add `requester=...` throughout the project.

Reply:
I was wondering the same thing when I went to add the new argument. This makes sense! thanks for the deeper explanation.